### PR TITLE
I2 overlapping ship

### PR DESCRIPTION
**Pull request by Adrlloyd**
Will add overlapping method to unsure no ship placements are invalid. test added to ensure this is correct.

### CODE REVIEW

**Review by Pocketzs — APPROVED** [review]

**Review by Pocketzs — APPROVED** [review]

**Review by Pocketzs — APPROVED** [review]

File: lib/board.rb [hunk 1]
```diff
@@ -68,10 +68,24 @@ def valid_placement?(ship, coordinates)
...
+    ships = coordinates.map do |coordinate|
+      cells[coordinate].ship
+    end
+    if ships.all? {|ship|  ship == nil}
+      true
+    else
+      false
+    end
+  end
+
```
Comment (Pocketzs) [inline lib/board.rb:81]: Good method to use within valid_placement method

**Review by Pocketzs — COMMENTED** [review]
Put overlap check into valid placement method

File: lib/board.rb (hunk 1 above)
Comment (Adrlloyd) [reply lib/board.rb:81]: Agreed this will resolve the error, and pass the test.

Comment (Pocketzs) [reply lib/board.rb:81]: Okay Ill approve this review

**Pull request merged**